pkg/storage/storage.go

```diff
@@ -174,3 +228,308 @@ func (s *svc) DeleteUploadID(ctx context.Context, user, bucket, object, uploadID
+		return fmt.Errorf("unable to decrement amount of completed consistency check tasks: %w", err)
+	}
+
+	return nil
+}
+
+func (s *svc) GetConsistencyCheckScheduledCounter(ctx context.Context, id string) (uint64, error) {
+	key := fmt.Sprintf("ccv:c:%s:scheduled", id)
+	count, err := s.client.Get(ctx, key).Uint64()
+	if errors.Is(err, redis.Nil) {
+		return 0, nil
+	}
+	if err != nil {
+		return 0, fmt.Errorf("unable to get amount of scheduled consistency check tasks: %w", err)
+	}
+
+	return count, nil
+}
+
+func (s *svc) GetConsistencyCheckCompletedCounter(ctx context.Context, id string) (uint64, error) {
+	key := fmt.Sprintf("ccv:c:%s:completed", id)
+	count, err := s.client.Get(ctx, key).Uint64()
+	if errors.Is(err, redis.Nil) {
+		return 0, nil
+	}
+	if err != nil {
+		return 0, fmt.Errorf("unable to get amount of scheduled consistency check tasks: %w", err)
+	}
+
+	return count, nil
+}
+
+func (s *svc) DeleteConsistencyCheckScheduledCounter(ctx context.Context, id string) error {
+	key := fmt.Sprintf("ccv:c:%s:scheduled", id)
+	if err := s.client.Unlink(ctx, key).Err(); err != nil {
+		return fmt.Errorf("unable to delete counter for scheduled consistency check tasks: %w", err)
+	}
+
+	return nil
+}
+
+func (s *svc) DeleteConsistencyCheckCompletedCounter(ctx context.Context, id string) error {
+	key := fmt.Sprintf("ccv:c:%s:completed", id)
+	if err := s.client.Unlink(ctx, key).Err(); err != nil {
+		return fmt.Errorf("unable to delete counter for completed consistency check tasks: %w", err)
+	}
+
+	return nil
+}
+
+func (s *svc) StoreConsistencyCheckID(ctx context.Context, id string) error {
+	cmd := s.client.SAdd(ctx, "ccv:id", id)
+	if cmd.Err() != nil {
+		return fmt.Errorf("unable to add id to consistency check set: %w", cmd.Err())
+	}
+
+	affected := cmd.Val()
+	if affected == 0 {
+		return errors.New("consistency check id already exists")
+	}
+
+	return nil
+}
+
+func (s *svc) DeleteConsistencyCheckID(ctx context.Context, id string) error {
+	cmd := s.client.SRem(ctx, "ccv:id", id)
+	if cmd.Err() != nil {
+		return fmt.Errorf("unable to delete from consistency check set: %w", cmd.Err())
+	}
+
+	return nil
+}
+
+func (s *svc) ListConsistencyCheckIDs(ctx context.Context) ([]string, error) {
+	res, err := s.client.SMembers(ctx, "ccv:id").Result()
+	if errors.Is(err, redis.Nil) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, fmt.Errorf("unable to list set: %w", err)
+	}
+
+	return res, nil
+}
+
+func (s *svc) AddToConsistencyCheckSet(ctx context.Context, record *ConsistencyCheckRecord) error {
+	key := fmt.Sprintf("ccv:s:%s:%s:%s", record.ConsistencyCheckID, record.Object, record.ETag)
+	if err := luaAddToConsistencySet.Run(ctx, s.client, []string{key}, record.Storage, record.StorageCount).Err(); err != nil {
+		return fmt.Errorf("unable to add object info to consistency check set: %w", err)
+	}
+
+	return nil
+}
+
+func (s *svc) FindConsistencyCheckSets(ctx context.Context, id string) ([]ConsistencyCheckResultEntry, error) {
```

Comment:
If i understand it correctly, this method should return not-matching objects. It can be done later in a separate PR but it would be nice to have a shortcut method returning if entries exists.

Reply:
Adding such method is not a problem at all. Will do.

Reply:
Added corresponding function HasConsistencyCheckSets.